language/move-vm/runtime/src/interpreter.rs

```diff
@@ -210,6 +230,28 @@ impl Interpreter {
         function: Arc<Function>,
         ty_args: Vec<Type>,
     ) -> VMResult<()> {
+        // TODO Remove once native functions push a frame on the stack
+        let module_id = match function.module_id() {
```

Comment:
should this go inside the `map_err`? I am not clear what is doing up here....

Reply:
I had some issues with Arc. I think I just need to clone to fix this